modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/DatabaseNodeService.java

```diff
@@ -150,8 +156,8 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc) {
             Files.createDirectories(geoipTmpDirectory);
         }
         LOGGER.debug("initialized database node service, using geoip-databases directory [{}]", geoipTmpDirectory);
-        ingestServiceArg.addIngestClusterStateListener(this::checkDatabases);
         this.ingestService = ingestServiceArg;
+        clusterService.addListener(event -> checkDatabases(event.state()));
```

Comment:
Shall we wait here or in `checkDatabase` for the cluster state to be recovered?

```
        if (event.state().blocks().hasGlobalBlock(GatewayService.STATE_NOT_RECOVERED_BLOCK)) {
            return;
        }
```

Reply:
Would be a good idea to early out in the case where the state is blocked. I'll add that for sanity